### PR TITLE
[1LP][RFR] Added Azure and Network credentials with Credentials modal page refactor.

### PR DESCRIPTION
**Pull request by sbulage**
This PR contains enhancement in Credentials Modal Page:

- [x] Refactored Credentials page: Redundant addition of Credentials `dict` in `create` and `update`.
- [x] Added `Azure` and `Network` Credentials.
- [x] Automated manual tests.

PRT Run: 
{{ pytest: cfme/tests/ansible/test_embedded_ansible_basic.py -k test_embedded_ansible_credential_crud -svvvv --long-running }}

### CODE REVIEW

File: cfme/ansible/credentials.py [hunk 1]
```diff
@@ -212,6 +252,97 @@ def before_fill(self, values):
                 continue
 
 
+def machine_credential(credential):
```
Comment (john-dupuy) [inline cfme/ansible/credentials.py]: _Optional_ refactor, shouldn't these all be plural? i.e. `machine_credentials(credentials)`?

Comment (sbulage) [reply cfme/ansible/credentials.py]: Okay I will update it.

File: cfme/ansible/credentials.py [hunk 2]
```diff
@@ -156,6 +156,46 @@ class CredentialFormGCEView(View):
...
+        username = Input(locator='.//input[@title="Username for this credential"]')
+        password = Input(locator='.//input[@title="Password for this credential"]')
+        authorize = Input(
+            locator='.//input[@title="Whether to use the authorize mechanism"]'
+        )
+        authorize_password = Input(
+            locator='.//input[@title="Password used by the authorize mechanism"]'
+        )
+        ssh_key = TextInput(
+            locator='.//textarea[@title="RSA or DSA private key to be used instead of password"][2]'
```
Comment (izapolsk) [inline cfme/ansible/credentials.py]: indexes look a bit frightening. isn't there a better way to uniquely identify this element ?

File: cfme/ansible/credentials.py [hunk 3]
```diff
@@ -156,6 +156,46 @@ class CredentialFormGCEView(View):
...
+        )
+        private_key_phrase = Input(
+            locator='.//input[@title="Passphrase to unlock SSH private key if encrypted"][2]'
```
Comment (izapolsk) [inline cfme/ansible/credentials.py]: same here

**Pull request merged**